configure.ac

```diff
@@ -63,16 +63,48 @@ AC_ARG_ENABLE([esapi],
                             [build the esapi layer (default is yes)])],
             [enable_esapi=$enableval],
             [enable_esapi=yes])
-AS_IF([test "x$enable_esapi" != xno],
+
+
+AC_ARG_ENABLE([esapi], 
+    AS_HELP_STRING([--disable-esapi], [Enable TSS ESYS API library @<:@default=yes@:>@]),
+    [],
+    [enable_esapi="yes"])
+AM_CONDITIONAL(ESAPI, test "x$enable_esapi" = "xyes")
+
+AC_ARG_WITH([crypto],
+            [AS_HELP_STRING([--with-crypto={gcrypt,ossl}],
+                            [sets the ESAPI crypto backend (default is gcrypt)])],
+            [],
+            [with_crypto=gcrypt])
+
+AM_CONDITIONAL(ESYS_OSSL, test "x$with_crypto" = "xossl")
+AM_CONDITIONAL(ESYS_GCRYPT, test "x$with_crypto" = "xgcrypt")
+
+AS_IF([test "x$with_crypto" != "xgcrypt"],
+    AS_IF([test "x$with_crypto" != "xossl"],
+          AC_MSG_ERROR([Bad value for --with-crypto $with_crypto])))
+
+AS_IF([test "x$enable_esapi" != xno -a "x$with_cryptoprovider" = "xgcrypt"],
       [AC_CHECK_HEADER([gcrypt.h],
                        [],
                        [AC_MSG_ERROR([Missing required header: gcrypt.h.])])])
-AS_IF([test "x$enable_esapi" != xno],
+
+AS_IF([test "x$enable_esapi" != xno -a "x$with_cryptor" = "xgcrypt"],
       [AC_CHECK_LIB([gcrypt],
                     [gcry_mac_open],
                     [],
                     [AC_MSG_ERROR([Missing required library: gcrypt.])])])
-AM_CONDITIONAL([ESAPI], [test "x$enable_esapi" != xno])
+
+AS_IF([test "x$enable_esapi" != xno -a "x$with_crypto" = "xossl"],
+      [AC_CHECK_HEADER([openssl/ssl.h],
+                       [],
+                       [AC_MSG_ERROR([Missing required header: openssl/ssl.h.])])])
+
+AS_IF([test "x$enable_esapi" != xno -a "x$with_crypto" = "xossl"],
+    AC_CHECK_LIB(ssl, OPENSSL_init_ssl, [FOUND_SSL_LIB="yes"]))
```

Comment:
Should we require a openssl version?
EVP_MD_CTX_create() and EVP_MD_CTX_destroy() were renamed to EVP_MD_CTX_new() and EVP_MD_CTX_free() in OpenSSL 1.1.

Reply:
EVP_MD_CTX_create and EVP_MD_CTX_destroy are still defined in openssl/evp.h for:
openssl  1.1.0f-3 
I have tested the crypto=ossl version with OpenSSL 1.1.0 and OpenSSL 1.0.2.